modules/optimization/src/utils/ParameterMesh.C

```diff
@@ -401,9 +401,17 @@ ParameterMesh::closestPoint(const Elem & elem, const Point & p) const
   }
 }
 
-Real
-ParameterMesh::computeGradientL2RegularizationObjective(
-    const std::vector<Real> & parameter_values) const
+template <typename T>
+T
+ParameterMesh::computeRegularizationLoop(
+    const std::vector<Real> & parameter_values,
+    RegularizationType /*reg_type*/,
+    const std::function<void(const std::vector<std::vector<Real>> &,
```

Comment:
Can you call the lambda from this function and not pass it?  Why does it have to be passed?